stdlib/src/collections/list.mojo

```diff
@@ -493,16 +493,67 @@ struct List[T: CollectionElement, hint_trivial_type: Bool = False](
         self.capacity = new_capacity
 
     fn append(mut self, owned value: T):
-        """Appends a value to this list.
+        """Appends a value to this list. If there is no capacity left, resizes
+        to twice the current capacity. Except for 0 capacity where it sets 1.
 
         Args:
             value: The value to append.
         """
-        if self.size >= self.capacity:
-            self._realloc(max(1, self.capacity * 2))
-        (self.data + self.size).init_pointee_move(value^)
+        if len(self) >= self.capacity:
+            self._realloc(self.capacity * 2 + int(self.capacity == 0))
+        (self.data + len(self)).init_pointee_move(value^)
         self.size += 1
 
+    fn append[
+        D: DType, //
+    ](mut self: List[Scalar[D], *_, **_], value: SIMD[D, _]):
+        """Appends a vector to this list. If there is no capacity left, resizes
+        to `len(self) + value.size`.
```

Comment:
**Suggestion** Same suggestion here re. splitting this up so that the first line of the description stays short and simple.